src/main/java/org/spongepowered/api/item/inventory/Inventory.java

```diff
@@ -318,23 +319,46 @@
     <V> Optional<V> getProperty(Property<V> property);
 
     /**
-     * Query this inventory for inventories matching any of the supplied
-     * queries. Logical <code>OR</code> is applied between operands.
+     * Query this inventory with given {@link Query}
      *
-     * @param operations queries to check against
-     * @return the query result
+     * @param query The query
+     *
+     * @return The queried inventory
+     */
+    Inventory query(Query query);
+
+    /**
+     * Query this inventory with given {@link QueryType} and {@code param}.
+     *
+     * @param type The query type
+     * @param param The query parameter
+     * @param <P> The query parameter type
+     *
+     * @return The queried inventory
+     */
+    <P> Inventory query(QueryType.OneParam<P> type, P param);
+
+    /**
+     * Query this inventory with given {@link QueryType} and {@code params}.
+     *
+     * @param type The query type
+     * @param param1 The first query parameter
+     * @param param2 The second query parameter
+     * @param <P1> The first query paramater type
+     * @param <P2> The second query parameter type
+     *
+     * @return The queried inventory
      */
-    Inventory query(QueryOperation<?>... operations);
+    <P1, P2> Inventory query(QueryType.TwoParam<P1, P2> type, P1 param1, P2 param2);
```

Comment:
Agree with ST-DDT. We can virtually have infinite number of params and the list of such methods can go on, but we don't want to explode the class with such methods